src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -0,0 +1,1021 @@
+        }
+        return value;
+    }
+
+    /**
+     * Returns the long value parsed parsed from a string range of up to sixteen hexadecimal
+     * characters.
+     * The characters in the range from {@code index} to {@code index + length - 1},
+     * inclusive, must be valid hex digits according to {@link #fromHexDigit(int)}.
+     * The {@code delimiter}, {@code prefix} and {@code suffix} are not used.
+     *
+     * @param string a CharSequence containing the characters
+     * @param index the index of the first character of the range
+     * @param length the number of hexadecimal digits to parse
+     * @return the value parsed from the string range
+     * @throws  IndexOutOfBoundsException if the sub-range is out of bounds
+     *          for the {@code CharSequence}
+     * @throws  IllegalArgumentException if {@code length} is greater than sixteen (16) or
+     *          if any of the characters is not a hexadecimal character
+     */
+    public long fromHexDigitsToLong(CharSequence string, int index, int length) {
+        Objects.requireNonNull(string, "string");
+        checkDigitCount(length, 16);
+        Objects.checkFromIndexSize(index, length, string.length());
+        long value = 0L;
+        for (int i = 0; i < length; i++) {
+            value = (value << 4) + fromHexDigit(string.charAt(index + i));
+        }
+        return value;
+    }
+
+    /**
+     * Returns {@code true} if the other object is a {@code HexFormat}
+     * with the same parameters.
+     *
+     * @param o an object, may be null
+     * @return {@code true} if the other object is a {@code HexFormat} and the parameters
+     *         {@code uppercase}, {@code delimiter}, {@code prefix}, and {@code suffix} are equal;
+     *         otherwise {@code false}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        HexFormat otherHex = (HexFormat) o;
+        return delimiter.equals(otherHex.delimiter) &&
+                prefix.equals(otherHex.prefix) &&
+                suffix.equals(otherHex.suffix) &&
+                Arrays.equals(digits, otherHex.digits);
+    }
+
+    /**
+     * Returns a hashcode for this {@code HexFormat} that is consistent with
+     * {@link #equals(Object) equals}.
+     *
+     * @return a hashcode for this {@code HexFormat}
+     */
+    @Override
+    public int hashCode() {
+        int result = Objects.hash(delimiter, prefix, suffix);
+        result = 31 * result + Boolean.hashCode(Arrays.equals(digits, UPPERCASE_DIGITS));
+        return result;
+    }
+
+    /**
+     * Returns a description of the formatter parameters for {@code uppercase},
+     * {@code delimiter}, {@code prefix}, and {@code suffix}.
+     *
+     * @return return a description of this {@code Formatter}
+     */
+    @Override
+    public String toString() {
```

Comment:
The caller may need to provide their own context for the output.

Reply:
I was thinking of the case where the caller invokes it by accident (due to incorrect code, e.g. `"..." + hexFormat` instead of `"..." + hexFormat.toHexDigits(...)`) and then wonders what "uppercase: true, ..." means or where it even comes from. For example `java.util.Optional` also includes "Optional[...]".